src/ast/passes/semantic_analyser.cpp

```diff
@@ -1058,7 +1058,7 @@ void SemanticAnalyser::visit(Call &call)
             << std::to_string(*sig)
             << " is not a valid signal, allowed range: [1,64]";
       }
-    } else if (arg.type.type != Type::integer) {
+    } else if (arg.type.GetTy() != Type::integer) {
```

Comment:
I realize it's a little out of scope on this PR, but would be a good opportunity to simplify some conditionals to `IsIntTy()` style here and below. A little more readable IMO.

Lemme know if you feel like taking that on - another PR is fine too if you want.

Reply:
Sure, I don't mind doing it in this PR. I've already updated all the callsites under tests/ - will extend that for the rest of the conditionals that would have used GetTy() as well.